RS.Fritz.Manager.UI/UserControls/Hosts/HostsViewModel.cs

```diff
@@ -18,17 +31,89 @@ public HostsViewModel(DeviceLoginInfo deviceLoginInfo, ILogger logger, IFritzSer
             get => hostsGetHostNumberOfEntriesResponse; set { _ = SetProperty(ref hostsGetHostNumberOfEntriesResponse, value); }
         }
 
+        public HostsGetHostListPathResponse? HostsGetHostListPathResponse
+        {
+            get => hostsGetHostListPathResponse; set { _ = SetProperty(ref hostsGetHostListPathResponse, value); }
+        }
+
+        public HostsGetHostListResponse? HostsGetHostListResponse
+        {
+            get => hostsGetHostListResponse; set { _ = SetProperty(ref hostsGetHostListResponse, value); }
+        }
+
+        public HostsGetGenericHostEntryResponse? HostsGetGenericHostEntryResponse
+        {
+            get => hostsGetGenericHostEntryResponse; set { _ = SetProperty(ref hostsGetGenericHostEntryResponse, value); }
+        }
+
         protected override async Task DoExecuteDefaultCommandAsync()
         {
             await Domain.TaskExtensions.WhenAllSafe(new[]
                 {
-                   GetHostsGetHostNumberOfEntriesAsync(),
+                    GetHostsGetHostListPathAsync(),
+                    GetHostsGetHostNumberOfEntriesAsync(),
+                    GetHostsGetGenericHostEntryAsync(index)
                 });
         }
 
         private async Task GetHostsGetHostNumberOfEntriesAsync()
         {
             hostsGetHostNumberOfEntriesResponse = await FritzServiceOperationHandler.GetHostsGetHostNumberOfEntriesAsync();
         }
+
+        private async Task GetHostsGetGenericHostEntryAsync(ushort index)
+        {
+            hostsGetGenericHostEntryResponse = await FritzServiceOperationHandler.GetHostsGetGenericHostEntryAsync(index);
+        }
+
+        private async Task GetHostsGetHostListPathAsync()
+        {
+            hostsGetHostListPathResponse = await FritzServiceOperationHandler.GetHostsGetHostListPathAsync();
+
+            if (HostsGetHostListPathResponse != null && FritzServiceOperationHandler.InternetGatewayDevice != null)
+            {
+                HostsGetHostListPathResponse.HostListPathLink = new Uri("http://" + FritzServiceOperationHandler.InternetGatewayDevice.PreferredLocation.Host + ":" + "49000" + hostsGetHostListPathResponse.HostListPath);
+                hostsGetHostListResponse = await FritzServiceOperationHandler.GetHostsGetHostListAsync(hostsGetHostListPathResponse.HostListPath);
+
+                using var stringReader = new StringReader(hostsGetHostListResponse.DeviceHostsListXml);
+                using var xmlTextReader = new XmlTextReader(stringReader);
+
+                try
+                {
+                    // fill the array with hosts in HostListResponse (not the actual Data Context)
```

Comment:
Can the code inside the try be replaced with these 2 lines?
hostsGetHostListResponse.DeviceHostsList = (DeviceHostsList?)new XmlSerializer(typeof(DeviceHostsList)).Deserialize(xmlTextReader);
hostsGetHostListPathResponse.DeviceHostsCollection = new ObservableCollection<DeviceHost>(hostsGetHostListResponse.DeviceHostsList?.DeviceHosts ?? Array.Empty<DeviceHost>());